src/Toml/Parser/TOML.hs

```diff
@@ -1,44 +1,46 @@
 module Toml.Parser.TOML
-       ( keyValP
+       ( hasKeyP
        , tableHeaderP
+       , inlineTableP
        , tomlP
        ) where
 
 import Control.Applicative (Alternative (many))
+import Control.Applicative.Combinators (sepEndBy, between, eitherP)
 
 import Toml.Parser.Core (Parser, sc, text)
-import Toml.Parser.Value (keyP, tableNameP, valueP)
+import Toml.Parser.Value (keyP, tableNameP, anyValueP)
 import Toml.PrefixTree (Key (..), fromList)
-import Toml.Type (AnyValue, TOML (..), typeCheck)
+import Toml.Type (AnyValue, TOML (..))
 
 import qualified Data.HashMap.Lazy as HashMap
 
+hasKeyP :: Parser (Key, Either AnyValue TOML)
+hasKeyP = (,) <$> keyP <* text "=" <*> eitherP anyValueP inlineTableP
 
-keyValP :: Parser (Key, AnyValue)
-keyValP = do
-    k    <- keyP
-    _    <- text "="
-    uval <- valueP
-    case typeCheck uval of
-        Left err -> fail $ show err
-        Right v  -> pure (k, v)
-
+inlineTableP :: Parser TOML
+inlineTableP = between (text "{") (text "}")
```

Comment:
I feel like `char '{'` can be more efficient here but it probably will work worse with consuming spaces after `{` so having `text` is okay for now 👍

Reply:
Speaking of which: the TOML specifications say "it is strongly discouraged to break an inline table onto multiples lines" and somewhere else I found the same info with even stronger language. Right now `inlineTableP` can parse inline tables with line breaks, and it would probably messy to change that. What do you think? 
I think it's probably fine to have the capability of parsing, as long as we are never pretty-printing an inline table on multiple lines.

Reply:
@jiegillet Sure, feel free to replace! It would be a good addition 👍

Reply:
> Right now inlineTableP can parse inline tables with line breaks, and it would probably messy to change that. What do you think?

I'm okay if `tomland` parses inline tables on multiple lines. And I understand that it will be hard to fix with existing code because spaces and line breaks are all the same for megaparsec and introducing distinction will be hard...